invenio_app_ils/permissions.py

```diff
@@ -35,6 +37,22 @@ def backoffice_permission(*args, **kwargs):
     return Permission(backoffice_access_action)
 
 
+def file_download_permission(obj):
+    """File download permissions."""
+    search = current_app_ils.eitem_search
+    for hit in search.search_by_bucket_id(str(obj.bucket_id)).scan():
+        if not hit.to_dict().get("open_access", False):
```

Comment:
@equadon This should be checked against ES6, no?

Reply:
The hit object didn't change much AFAIK from 6 to 7 so it should be okay but we'll see when the ES6 tests are running.

Reply:
I tested it both in ES6 and ES7 and `to_dict()` is supported. I'm fine with using the DB approach instead.

Reply:
Since we don't use Records-Files and only rely on Files-REST we can't do this.

Reply:
True! Let's change it as discussed together :)